tests/Integration/Behaviour/Features/Context/Domain/ImageSettings/ImageTypeContext.php

```diff
@@ -0,0 +1,216 @@
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\ImageSettings;
+
+use Behat\Gherkin\Node\TableNode;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\AddImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\BulkDeleteImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\DeleteImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\EditImageTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Query\GetImageTypeForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\QueryResult\EditableImageType;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\ValueObject\ImageTypeId;
+use Tests\Integration\Behaviour\Features\Context\Domain\AbstractDomainFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class ImageTypeContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I create an image type :imageTypeName with following properties:
+     */
+    public function createImageTypeUsingCommand(string $imageTypeName, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+
+        $command = new AddImageTypeCommand(
+            $imageTypeName,
+            $data['width'],
+            $data['height'],
+            $data['products'],
+            $data['categories'],
+            $data['manufacturers'],
+            $data['suppliers'],
+            $data['stores']
+        );
+
+        /** @var ImageTypeId $imageTypeId */
+        $imageTypeId = $this->getCommandBus()->handle($command);
+
+        $this->getSharedStorage()->set($imageTypeName, $imageTypeId->getValue());
+    }
+
+    /**
+     * @When I edit image type :imageTypeName with following properties:
+     */
+    public function editImageTypeUsingCommand(string $imageTypeName, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+        $this->getSharedStorage()->exists($imageTypeName);
+
+        $command = new EditImageTypeCommand($this->getSharedStorage()->get($imageTypeName));
+        $command->setName($imageTypeName);
+        $command->setWidth($data['width']);
+        $command->setHeight($data['height']);
+        $command->setProducts($data['products']);
+        $command->setCategories($data['categories']);
+        $command->setManufacturers($data['manufacturers']);
+        $command->setSuppliers($data['suppliers']);
+        $command->setStores($data['stores']);
+
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I delete image type :imageTypeName.
+     */
+    public function deleteImageTypeUsingCommand(string $imageTypeName)
+    {
+        $this->getSharedStorage()->exists($imageTypeName);
+        $imageTypeId = $this->getSharedStorage()->get($imageTypeName);
+        $command = new DeleteImageTypeCommand($imageTypeId);
```

Comment:
```suggestion
        $command = new DeleteImageTypeCommand($this->getSharedStorage()->get($imageTypeName));
```
Just a suggestion, but since the `get` method already checks if the key exist this code could be simplified (same suggestion for other similar use cases in this PR, but also optional) especially since `exists` simply return a bool and does not trigger an exception so calling it like this doesn't bring anything